File: day01/day01.nim
import std/strutils
import std/sequtils
import std/algorithm
import std/sugar

let filename = "./input"
var calorieSums = collect:
  for item in filename.readFile.split("\n\n"):
    item.splitLines.map(parseInt).foldl(a + b)

calorieSums.sort()

echo calorieSums[calorieSums.high]
echo calorieSums[calorieSums.high] + calorieSums[calorieSums.high - 1] + calorieSums[calorieSums.high - 2]

## Answers
## 70296
## 205381

File: day01/day01.go
package main

import (
	"bytes"
	"fmt"
	"log"
	"os"
	"sort"
	"strconv"
)

func check(err error) {
	if err != nil {
		log.Fatalln(err)
	}
}

func main() {
	filename := "./input"
	body, err := os.ReadFile(filename)
	check(err)

	var calorieSums []int
	res := bytes.Split(body, []byte("\n\n"))
	for _, it := range res {
		var sum int
		for _, i := range bytes.Split(it, []byte("\n")) {
			num, err := strconv.Atoi(string(i))
			check(err)
			sum += num
		}
		calorieSums = append(calorieSums, sum)
	}

	sort.Ints(calorieSums)

	fmt.Println(calorieSums[len(calorieSums)-1])
	fmt.Println(calorieSums[len(calorieSums)-1] + calorieSums[len(calorieSums)-2] + calorieSums[len(calorieSums)-3])
}
